docusaurus/docs/iOS/client/controllers/channels.md

```diff
@@ -55,28 +55,73 @@ let compoundFilter: Filter<ChannelListFilterScope> = .and([
 The `query.sort` is an array of sorting options. Sorting options are applied based on their order in the array so the first option has the highest impact while the others are used mainly as a tiebreakers. By default, the channel list is sorted by `updated_at`.
 
 ### 2. Create a controller
+The simplest way to create a controller is by using the static method `channelListController(query:)` on your `ChatClient`.
+```swift
+let controller = ChatClient.shared.channelListController(query: query)
+```
+By default, the SDK will automatically handle filtering the channels as they get created. Whenever there is a web socket event that a channel has been created, the SDK will only insert it in the channel list if it matches the query.
+
+:::note
+In cases though, where the query provided contains, extra data or custom filters, **the SDK may not be able to automatically match the filter query**. In this case, you will need to provide a filtering closure.
+:::
+
+#### Filtering with extra data
+First we need to disable the Channel auto-filtering. We can do that by turning the `isChannelAutomaticFilteringEnabled` in your `ChatClient` configuration, to `false`.
+```swift
+extension ChatClient {
+    static let shared: ChatClient = {
+        // You can grab your API Key from https://getstream.io/dashboard/
+        var config = ChatClientConfig(apiKeyString: "<# Your API Key Here #>")
+        config.isChannelAutomaticFilteringEnabled = false
+
+        // Create an instance of the `ChatClient` with the given config
+        let client = ChatClient(config: config)
 
-The controller can be created using `ChatClient`:
+        return client
+    }()
+}
+```
+Then, we will need to provide to our ChannelController a filtering closure. We can achieve this with the code below:
 ```swift
 let controller = ChatClient.shared.channelListController(query: query, filter: { channel in
-    // the filtering logic goes here.
-    return true
+    // We are filtering for channels that:
+    // 1. The current is user is a member
+    // 2. A value exists for the extraData key `myCustomBooleanKey` and this value is `true`
+    return channel.membership != nil 
+        && channel.extraData["myCustomBooleanKey"] as? Bool == true
 })
 ```
+#### Filtering with custom filters
```

Comment:
I actually don't think we have any section in the docs on how to create custom filters 🤔 So maybe we should discard this one for now. Otherwise, we need to make an introduction here on how to create one.